doc/relevancy.md

```diff
@@ -0,0 +1,272 @@
+'sprint' or 'gallop' in them."
+
+It must be applied both at index time and at query time, to ensure that query
+terms are mapped to the same single value that exists in the index.
+
+### Filtering
+
+Additional configuration is defined in the elasticsearch_schema.yml and
+stems.yml files.  This configuration is merged with the JSON configuration,
+and then passed to elasticsearch directly.
+
+This blog post suggests that as well as using keywords, it can be useful to use
+'keep words' to only filter the phrases we want.
+
+The post suggests to first generate shingles, followed by synonyms, followed
+by keepwords.
+
+In other words: generate candidate keyphrases by shingling, expand them with
+synonyms, then cull out any non-synonyms with keepwords.
+
+
+## Popularity boosting
+
+How does it work?
+
+Popularity indicator is used to bias search results. It is one of the factors
+involved in weighting search results. A higher number indicates more pageviews.
+It is updated nightly.
+
+Popularity is derived from recent page views. The gov.uk search uses page
+popularity information extracted from Google. The data is gathered by the
+search-analytics application. That data is imported into Search API.
+
+### Popularity score
+
+`popularity_score` seems to be set in `PopularityLookup` class.
+
+```ruby
+popularity_score = 1.0 / (ranks[link] + SearchConfig.popularity_rank_offset)
+```
+
+Initial reasoning for the numbers (May 2014):
+
+"This field is populated from the page-traffic index, which in turn is
+populated from a dump file produced by the python scripts in the
+search-analytics repository.
+
+The popularity is currently defined here as 1.0 / the rank of the page in
+terms of traffic.  ie, Our highest traffic page gets 1.0, the next highest
+gets 0.5, the next highest gets 0.333.  This is a nice and stable measure."
+
+https://github.com/alphagov/search-api/commit/6cbd84f36ff70ce1be6a368e807c72ba09c74f23
+
+### Boosting the popularity score
+
+If we want to boost the popularity of a query then we do so by 0.001
+(POPULARITY_OFFSET = 0.001 in popularity.rb). It is not clear where this number
+came from.
+
+We also boost using boost_mode: multiply (Multiply the `_score` with the
+function result (default)).
+
+### Relative ranking
+
+The number of page views a link has over a 14 day period is stored in its
+`vc_14` field, and the rank relative to other pages on GOV.UK is stored in the
+`rank_14` field.
+
+It may be better to change rummager to use `vc_14` directly for its popularity
+boost, which would let us do partial popularity updates. This is a bigger
+change than we want to make right now, as it would need careful measurement
+to ensure we don't break queries.
+
+https://github.com/alphagov/search-api/blob/master/doc/arch/adr-005-incremental-popularity-updates.md
+
+
+## Query DSL usage
+
+Here is a map of what goes into our elasticsearch query: https://www.draw.io/#G0B3LRehFrM0QuaVRxV0hiZzhTbG8
```

Comment:
This diagram is inaccurate, we don't do shingling any more (and possibly other problems).  We should remake it to be higher res and also accurate.